campari/simulation.py

```
@@ -70,6 +70,16 @@ def simulate_images(image_list, diaobj,
     galra: float, the RA of the galaxy.
     galdec: float, the DEC of the galaxy.
     """
+    assert image_list is not None, "You must provide a list of images to simulate."
+    assert diaobj is not None, "You must provide a diaobj to simulate."
+    assert len(image_list) > 0, "You must provide at least one image to simulate."
+    assert bg_gal_flux is not None, "You must provide a background galaxy flux to simulate."
+    assert roman_path is not None, "You must provide a path to the Roman TDS files."
+    assert do_xshift is not None, "You must provide a value for do_xshift."
+    assert do_rotation is not None, "You must provide a value for do_rotation."
+    assert use_roman is not None, "You must provide a value for use_roman."
+    assert deltafcn_profile is not None, "You must provide a value for deltafcn_profile."
+    assert noise is not None, "You must provide a value for noise."
```

I strongly discourage `asserts` in code.  It terminates the running.  I would much prefer raising an error.  Those you can catch and decide what to do with.  

I understand why this happened.  When you shift to all arguments being keywords, it means that all arguments have defaults and are thus optional.  That's a tradeoff.  But `assert` is not the way I would do this.

I also feel like all of these `assert`s are somewhat redundant.  If any of them are `None` then when you try to use them you'll get basically the same error message raised (probably a `TypeError` complaining about `NoneType`) as these assert message provide.

`asserts` are fine and appropriate in tests, of course.

---

Replaced with raising an error! I could let them fail but this is at least clearer why they fail.